src/scrapers/main.py

```diff
@@ -31,22 +36,34 @@ def docs():
         print('{: <16}{}'.format(s.code, s.name))
 
 
-(opts, args) = parser.parse_args()
+arguments = parser.parse_args()
 
-if opts.help:
+if arguments.help:
     docs()
 else:
-    print(f'Running {len(args)} scrapers. Use --help to see all individual scrapers')
+    print(f'Running {len(arguments.scrapers)} scrapers. Use --help to see all individual scrapers')
+    if arguments.group:
+        if arguments.group != InitiativeGroup.SUPPLY \
+                and arguments.group != InitiativeGroup.DEMAND:
+            print(f"Invalid group given {arguments.group}")
+            exit()
+
+    run_scrapers = None
+    if arguments.scrapers:
+        run_scrapers = reduce(lambda a, b: a + b, arguments.scrapers)
+
     for scraper in scrapers:
-        if opts.limit:
-            scraper.limit = opts.limit
+        if arguments.limit:
+            scraper.limit = arguments.limit
+        if arguments.group:
+            scraper.set_group(InitiativeGroup.DEMAND if arguments.group == "demand" else InitiativeGroup.SUPPLY)
```

Comment:
`scraper.set_group(arguments.group)` moet nu gewoon kunnen aangezien je choices gebruikt
je zou ook het argument optional kunnen maken en dan een default setten

Reply:
Komt door mijn strong typed achtergrond en omdat ik een typehint in die set_group had gezet. Ik heb schijnbaar een andere verwachting bij typehints. ;) Het is inderdaad overbodig.